File: src/test/java/testdata/java/StaticFieldPOJO.java
package testdata.java;


import lombok.Data;

import java.io.Serializable;

@Data
public class StaticFieldPOJO implements Serializable {

    private static final long serialVersionUID = 0L;


    public static int thisStatic = 0;

    private final int thisFinal = 0;

    private transient int thisTransient = 0;


    private static final int thisStaticFinal = 0;

    private static transient int thisStaticTransient = 0;


    private final transient int thisFinalTransient = 0;

    private static final transient int thisStaticFinalTransient = 0;

}

File: src/test/java/testdata/kotlin/StaticFieldPOJO.kt
package testdata.kotlin

import lombok.Data
import java.io.Serializable

@Data
class StaticFieldPOJO : Serializable {
    private val thisFinal = 0

    @Transient
    private val thisTransient = 0

    @Transient
    private val thisFinalTransient = 0

    companion object {
        private const val serialVersionUID = 0L
        var thisStatic = 0
        private const val thisStaticFinal = 0

        @Transient
        private val thisStaticTransient = 0

        @Transient
        private val thisStaticFinalTransient = 0
    }
}